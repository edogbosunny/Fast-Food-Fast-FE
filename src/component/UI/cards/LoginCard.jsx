import React from 'react';
import '../../../styles/style.css';


const LoginCard = () => (

  <div className="Aligner ">
    <div className="loginCard ">
      <div className="aligncnt ">
        <h2 className="mt">Login To Your Account </h2>
        { /* <!-- <h4 style="margin-top:-25px; color:#f2f2f2 ">
    Get your food in minutes</h4> --> */}
        <form>
          <div className="divwidth ">
            <div>
              <input className="txtwidth form-style-1 " id="email" type="email" name="email" placeholder="Email " />
              <p className="ErrMsg" id="emailErrMsg" />
            </div>
            <div>
              <input
                className="txtwidth form-style-1"
                type="password"
                id="pass"
                name="password"
                minLength="5"
                placeholder="Password"
                required
              />
              <p className="ErrMsg" id="pwdErrMsg" />
              <br />
              <p className="ErrMsg" id="signinErrMsg" />
            </div>
            <div className="mg-top">
              <button type='button' href="" id="submitbtn" className="a5 ">
                    Login
              </button>
            </div>
          </div>

          <div className="bm" />
        </form>
      </div>
    </div>
  </div>
);

export default LoginCard;
