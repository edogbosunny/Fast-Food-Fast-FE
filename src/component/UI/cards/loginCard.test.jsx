import React from 'react';
import { mount, shallow } from 'enzyme';
import { LoginCard, mapStateToProps } from './LoginCard';
import {
  props, mockedLoginState, nextProps,
  Loginstate,
} from '../../../__mocks__/loginUtils';

describe('Login Container', () => {
  const wrapper = mount(
    <LoginCard {...props} />,
  );

  it('should mount the login page Yeah!', () => {
    expect(wrapper).toMatchSnapshot();
  });

  it('should check if the errors prop exist', () => {
    // const wrapper = mount(<LoginCard {...props} />);
    expect(wrapper.props().errors.isAuthenticated).toBe(false);
  });
  it('should test mapStateToProps', () => {
    const state = mapStateToProps(mockedLoginState);
    expect(state).toBeTruthy();
  });

  it('should test componentwillrecieveprops()', () => {
    const componentWillReceivePropsSpy = jest.spyOn(LoginCard.prototype, 'componentWillReceiveProps');
    const shallowWrapper = shallow(<LoginCard {...props} />);
    shallowWrapper.instance().componentWillReceiveProps(nextProps);
    expect(componentWillReceivePropsSpy).toHaveBeenCalledTimes(1);
  });

  it('should handle input change event for email', () => {
    const event = {
      target: { name: 'email', value: 'test@test.com' },
      preventDefault: () => {},
    };

    wrapper.instance().onChange(event);
    expect(wrapper.state().email).toEqual('test@test.com');
    expect(wrapper.instance().state.email).toEqual(event.target.value);
  });

  it('should test the onSubmit function', () => {
    const fakeEvent = { preventDefault: () => console.log('preventDefault') };
    const loginForm = wrapper.find('form');
    loginForm.simulate('submit', fakeEvent);
    const loginUser = jest.fn(Loginstate);
    const promise = new Promise((resolve) => {
      resolve(wrapper.instance().onSubmit);
    });
    promise.then(() => expect(loginUser).toHaveBeenCalledTimes(1));
  });
});
