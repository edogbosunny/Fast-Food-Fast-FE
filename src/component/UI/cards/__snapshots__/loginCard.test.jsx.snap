// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Login Container should mount the login page Yeah! 1`] = `
<LoginCard
  auth={
    Object {
      "isAuthenticated": true,
      "loading": null,
      "user": Object {
        "exp": 1549525677,
        "iat": 1549439277,
        "id": 1,
      },
    }
  }
  errors={
    Object {
      "email": "",
      "isAuthenticated": false,
      "loading": null,
      "message": "",
      "password": "",
    }
  }
  history={
    Object {
      "action": "PUSH",
      "push": [MockFunction] {
        "calls": Array [
          Array [
            "/",
          ],
        ],
        "results": Array [
          Object {
            "isThrow": false,
            "value": undefined,
          },
        ],
      },
    }
  }
  location={
    Object {
      "hash": "",
      "pathname": "/login",
      "search": "",
    }
  }
  loginUser={[MockFunction]}
  match={
    Object {
      "isExact": true,
      "path": "/login",
      "url": "/login",
    }
  }
  user={Object {}}
>
  <div
    className="Aligner "
  >
    <div
      className="loginCard "
    >
      <div
        className="aligncnt "
      >
        <h2
          className="mt"
        >
          Login To Your Account 
        </h2>
        <form
          noValidate={true}
          onSubmit={[Function]}
        >
          <div
            className="divwidth "
          >
            <div>
              <TextFieldGroup
                error=""
                minLength="5"
                name="email"
                onChange={[Function]}
                placeholder="Email Address"
                type="text"
                value=""
              >
                <div
                  className="txtwidth form-style-1"
                >
                  <input
                    className="form-control"
                    minLength="5"
                    name="email"
                    onChange={[Function]}
                    placeholder="Email Address"
                    required={true}
                    type="text"
                    value=""
                  />
                </div>
              </TextFieldGroup>
              <TextFieldGroup
                error=""
                name="password"
                onChange={[Function]}
                placeholder="Password"
                type="password"
                value=""
              >
                <div
                  className="txtwidth form-style-1"
                >
                  <input
                    className="form-control"
                    name="password"
                    onChange={[Function]}
                    placeholder="Password"
                    required={true}
                    type="password"
                    value=""
                  />
                </div>
              </TextFieldGroup>
              <small />
            </div>
            <div>
              <input
                className=" submitbtn btn btn-info btn-block mt-4"
                type="submit"
              />
            </div>
          </div>
          <div
            className="bm"
          />
        </form>
      </div>
    </div>
  </div>
</LoginCard>
`;
